test_automation/work_order_utility.py

```diff
@@ -0,0 +1,171 @@
+    elif "result" in check_submit.keys():
+        if response["result"]["code"] == check_submit["result"]["code"]:
+            err_cd = 0
+            logger.info("SUCCESS: WorkOrderSubmit - Request submitted and received expected error code")
+        else:
+            err_cd = 1
+            logger.info("ERROR: Failed at WorkOrderSubmit - Request submitted and not received expected result code.")
+    else:
+        logger.info("ERROR: Failed at WorkOrderSubmit - No Error or Result in response")
+        err_cd = 1
+
+    return err_cd
+
+def create_work_order_get_result(work_order_id):
+    """ Function to create work order get result request. """
+
+    logger.info("------------------Constructing WorkOrderGetResult------------------")
+    # create work order get result request
+    input_workorder_getresult = input_data = '{"jsonrpc": "2.0","method": "WorkOrderGetResult","id": 11,"params": {"workOrderId": ""}}'
+
+    input_json_temp = json.loads(input_workorder_getresult)
+    input_json_temp["params"]["workOrderId"] = work_order_id
+    input_json_str1 = json.dumps(input_json_temp)
+
+    return input_json_str1
+
+def process_work_order_get_result(work_order_id, uri_client):
+    """ Function to process work order get result response. """
+
+    # process work order get result and retrieve response
+    input_json_str1 = create_work_order_get_result(work_order_id)
+    logger.info("------------------Validating WorkOrderGetResult Response------------------")
+    response = {}
+    output_json_file_name = 'work_order_get_result'
+
+    while("result" not in response):
+        if "error" in response:
+            if response["error"]["code"] != 5:
+                logger.info("ERROR: Failed at WorkOrderGetResult - Response received with error code.")
+                err_cd = 1
+                break
+
+        response = process_request(uri_client, input_json_str1, output_json_file_name)
+        time.sleep(3)
+
+    return response
+
+def process_work_order(input_json_file, output_json_file_name, worker_obj, sig_obj, uri_client, private_key, err_cd, check_submit, check_get_result):
+    """ Function to process work order and perform basic validation of return code in response. """
+
+    if err_cd == 0:
+        # read json input file for the test case
+        logger.info("------------------Input file name: %s ---------------\n", input_json_file)
+        input_json_str1 = futils.read_json_file(input_json_file)
+        #----------------------------------------------------------------------------------
+        logger.info("------------------Testing WorkOrderSubmit------------------")
+
+        # create work order request
+        input_json_str1, work_order_id = create_work_order_request(input_json_str1, worker_obj)
+        # sign work order request
+        input_json_str1, session_iv, encrypted_session_key = sign_work_order_request(input_json_str1, worker_obj, sig_obj, private_key)
```

Comment:
This line is too long and is hard to read. Please limit lines to 80 characters.

Reply:
Fixed line length to below 80